src/components/structures/RoomSubList.js

```diff
@@ -126,6 +127,74 @@ const RoomSubList = createReactClass({
         }
     },
 
+    onHeaderKeyDown: function(ev) {
+        switch (ev.key) {
+            case Key.TAB:
+                // Prevent LeftPanel handling Tab if focus is on the sublist header itself
+                ev.stopPropagation();
+                break;
+            case Key.ARROW_LEFT:
+                // On ARROW_LEFT collapse the room sublist
+                if (!this.state.hidden && !this.props.forceExpand) {
+                    this.onClick();
+                }
+                ev.stopPropagation();
+                break;
+            case Key.ARROW_RIGHT: {
+                ev.stopPropagation();
+                if (this.state.hidden && !this.props.forceExpand) {
+                    // sublist is collapsed, expand it
+                    this.onClick();
+                } else if (!this.props.forceExpand) {
+                    // sublist is expanded, go to first room
+                    let element = document.activeElement;
```

Comment:
I wonder if you could avoid the !decending code path below and usage of `document.activeElement` if you did `ev.target.closest(".mx_RoomSubList_labelContainer")` here?

Reply:
That certainly would clean this up, will test

Reply:
ended up simplifying even more